src/GslibIO.jl

```diff
@@ -52,28 +81,42 @@ Optionally set the value used for missings `na`.
 """
 function load_legacy(filename::AbstractString, dims::NTuple{3,Int};
                      origin=(0.,0.,0.), spacing=(1.,1.,1.), na=-999)
-  open(filename) do fs
-    # skip header
-    readline(fs)
+  spec = parse_legacy(filename)                     
 
-    # number of properties
-    nvars = parse.(Int, readline(fs))
+  # handle missing values
+  replace!(spec.data, na=>NaN)
 
-    # properties names
-    vars = [Symbol(readline(fs)) for i in 1:nvars]
+  # create data dictionary
+  table = (; zip(spec.varnames, eachcol(spec.data))...)
+  domain = RegularGrid(dims, origin, spacing)
 
-    # read property values
-    X = readdlm(fs)
+  georef(table, domain)
+end
 
-    # handle missing values
-    replace!(X, na=>NaN)
+"""
+    load_legacy(filename, coordnames, na=-999)
 
-    # create data dictionary
-    data = (; zip(vars, eachcol(X))...)
-    domain = RegularGrid(dims, origin, spacing)
+Load legacy GSLIB `filename` into a PointSet using the properties in `coordnames` as coordinates.
+Optionally set the value used for missings `na`.
+"""
+function load_legacy(filename::AbstractString, coordnames=(:x, :y, :z); na=-999)
+  spec = parse_legacy(filename)
 
-    georef(data, domain)
-  end
+  @assert coordnames ⊆ spec.varnames && !isempty(coordnames) "invalid coordinate names"
+
+  # handle missing values
+  replace!(spec.data, na=>NaN)
+
+  # we need to identify and separate coordinates and actual attributes
+  coordinds = indexin(collect(coordnames), spec.varnames)
+  coords = spec.data[:,coordinds]'
+
+  # create table with varnames not in coordnames
+  attrinds = setdiff(1:length(spec.varnames), coordinds)
+  attrnames = spec.varnames[attrinds]
+  table = (; zip(attrnames, eachcol(spec.data[:,attrinds]))...)
+
+  georef(table, PointSet(coords))
```

Comment:
```suggestion
  table = (; zip(attrnames, eachcol(spec.data[:,attrinds]))...)
  domain = PointSet(coords)

  georef(table, domain)
```

Reply:
Confused about using o not using intermediate variables, such as `domain` in line 118. You have asked to remove such variables but you are now asking to use them. Could you please clarify this for future contribution?

Reply:
Of course. In this case the consistency of the code takes precedence. If you check the other branch with regular grid, you will see that the domain was created explicitly as well. Avoid intermediate variables whenever possible, but try to be consistent overall. There is no strict rule, it is case by case I guess.

Reply:
OK, I will try to apply a good criterion as much as possible.

Reply:
Thanks, that helps. Just thinking actively about it is a good thing.

Reply:
I think this one was not committed, can you please commit the change before we can merge?

Reply:
It seems the commit was not included somehow in the last push.

> Thanks, that helps. Just thinking actively about it is a good thing.

It is